### PR TITLE
Do not reset IPs in Den teardown

### PR DESCRIPTION
**Pull request by mkandler**

### CODE REVIEW

**Comment by mkandler** [comment]
* **#1512** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1512?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1512?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by jlewitt1 — APPROVED** [review]

File: runhouse/resources/hardware/cluster_factory.py [hunk 1]
```diff
@@ -292,6 +296,7 @@ def ondemand_cluster(
     instance_type: Optional[str] = None,
     num_nodes: Optional[int] = None,
     provider: Optional[str] = None,
+    cluster_status: Optional[ClusterStatus] = None,
```
Comment (jlewitt1) [inline runhouse/resources/hardware/cluster_factory.py:299]: can we add this param to the docstrings?

File: runhouse/resources/hardware/on_demand_cluster.py [hunk 2]
```diff
@@ -137,6 +138,7 @@ def __init__(
         self._docker_user = None
         self._namespace = kwargs.get("namespace")
         self._context = kwargs.get("context")
+        self.cluster_status = kwargs.get("cluster_status")
```
Comment (jlewitt1) [inline runhouse/resources/hardware/on_demand_cluster.py:141]:
```suggestion
        self._cluster_status = kwargs.get("cluster_status")
```

File: runhouse/resources/hardware/on_demand_cluster.py [hunk 3]
```diff
@@ -390,8 +406,32 @@ def _start_ray_workers(self, ray_port, env_vars):
 
         time.sleep(5)
 
+    def _update_cluster_status_from_sky_status(self, sky_status: str):
+        if sky_status == "UP":
+            self.cluster_status == ClusterStatus.RUNNING
```
Comment (jlewitt1) [inline runhouse/resources/hardware/on_demand_cluster.py:411]: shouldn't these be with a single `=` :
```self._cluster_status = ClusterStatus.RUNNING```

Comment (mkandler) [reply runhouse/resources/hardware/on_demand_cluster.py:411]: Oh wow, yeah definitely 🤦

File: runhouse/resources/hardware/on_demand_cluster.py [hunk 4]
```diff
@@ -390,8 +406,32 @@ def _start_ray_workers(self, ray_port, env_vars):
 
         time.sleep(5)
 
+    def _update_cluster_status_from_sky_status(self, sky_status: str):
+        if sky_status == "UP":
```
Comment (jlewitt1) [inline runhouse/resources/hardware/on_demand_cluster.py:410]: one thought to explicitly use sky values here, we could do something like:
```
from sky import ClusterStatus as SkyClusterStatus
if sky_status == SkyClusterStatus.UP
```

**Review by jlewitt1 — APPROVED** [review]

File: runhouse/resources/hardware/on_demand_cluster.py (hunk 4 above)
Comment (mkandler) [reply runhouse/resources/hardware/on_demand_cluster.py:410]: Good call

**Pull request merged**